File: GlobalConstStruct/src/main/C/demolib.c
#include <string.h>
#include <stdio.h>

struct Point {
  int x, y;
};

const struct Point mypoint = {.x=10,.y=20};
struct Point * translate(struct Point const * pt, int dx, int dy) {
    printf("pt.x: %d\n\r", pt->x);
    printf("pt.y: %d\n\r", pt->y);
    printf("dx: %d\n\r", dx);
    printf("dy: %d\n\r", dy);

    return pt;
}

File: GlobalConstStruct/src/main/java/eu/doppel_helix/dev/demoglobalconststruct/DemoLib.java
package eu.doppel_helix.dev.demoglobalconststruct;

import com.sun.jna.Library;
import com.sun.jna.Native;
import com.sun.jna.NativeLibrary;
import com.sun.jna.Pointer;
import com.sun.jna.Structure;

public interface DemoLib extends Library {
    NativeLibrary LIBRARY = NativeLibrary.getInstance("demolib");
    DemoLib INSTANCE = Native.load("demolib", DemoLib.class);

    @Structure.FieldOrder({"x", "y"})
    class Point extends Structure {

        public volatile int x, y;

        public Point() {
        }

        public Point(Pointer p) {
            super(p);
            read();
        }
    }

    Point translate(Point pt, int x, int y);
}